### DISCUSSION TITLE
Unable to set sensitivity on Neo Coolcam motion sensor NAS_PD02Z-2

### DISCUSSION THREAD

**Discussion by Mastiffen** [zwave-js/zwave-js#7484] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
When trying to update the sensitivity parameter it stands "chewing" until it fails.

What did you expect to happen instead?
Values should be updated.

  1. Go to configuration V1
  2. Click on Sensitivity value
  3. Change the setting
  4. Click on the update arrow

Anything else we should know?

### Software versions

Driver (node-zwave-js): ...

Z-Wave JS UI: 9.27.7.d3d7049

Home Assistant Z-Wave Integration: N/A

Home Assistant Z-Wave JS Addon:  N/A.

ioBroker.zwave2 Adapter: 

If you are using something non-standard, tell us here: ...

### Z-Wave Controller (Stick/Dongle/...)

Aeotec Z-Stick Gen 5 Firmware 1.0

### Device information

Manufacturer: Shenzhen Neo Electronics Co., Ltd.
Model name: Motion Sensor NAS_PD02Z-2
Node ID: 62

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [ ] The log includes the problematic interaction with the device _(if applicable)_.

- [ ] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwave-js-ui-store.zip](https://github.com/user-attachments/files/18109814/zwave-js-ui-store.zip)

**Comment by Mastiffen** [comment]
I have tried different settings between 1 and 99 (which is the interval that the GUI gives me), but nothing works. I have a sneaking feeling that the problem is that the parameter is not the one it says it is. The motion sensor is unusable as it is, because it far to sensitive and triggers on nothing.

**Comment by Mastiffen** [comment]
I just remembered that I had another one of those, slight difference, this one has a magnetic mount. I took that out of the drawer, and it has the exact same problem.

**Comment by kpine** [comment]
It's a battery device, did you wake it up to apply the configuration change?

**Reply by Mastiffen** [reply]
Thanks for answering! And yes, of course. I didn't mention it, but I have been using Z-Wave for about 8 years, so I'm not that much of a newbie. ;) I think I didn't explain it correctly, when I wake it up instead of waiting for the regular "check-in" will make it fail at once.

**Reply by kpine** [reply]
I think you'll need to provide driver logs as requested then. There are no driver logs in your attachment.

**Reply by Mastiffen** [reply]
Sorry, I have not ever messed with the logs before. Are these the correct ones?
[zwave-js-ui-store.zip](https://github.com/user-attachments/files/18116041/zwave-js-ui-store.zip)

**Reply by kpine** [reply]
Nope. https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs

**Reply by Mastiffen** [reply]
I tried to follow that, yeah. Here's another attempt, it seemed like the level setting to debug didn't take the first time, sorry. It was still on info. I set it to debug, restarted and tried the setting that doesn't work. Am I missing any files in this, and if so what?
[zwave-js-ui-store (2).zip](https://github.com/user-attachments/files/18116306/zwave-js-ui-store.2.zip)

**Reply by Mastiffen** [reply]
It's not the only one that fails either, several other settings fail when I try to change them. Some work, others do not. If I try to read those in the Custom Configuration "Get" they all come up as "Undefined". The working ones come up as the value they are set to. So that makes me think there is something wrong in the parameter numbering.

**Reply by kpine** [reply]
Driver logs are still missing.

![image](https://github.com/user-attachments/assets/347e0562-57c1-4955-bbd9-c01ea2a4940b)

**Reply by Mastiffen** [reply]
Finally found it! The setup is different, I guess it's because it' another version. This is what it looks like here. I had to activate log to file there as well, not just in the general section. My bad. 
![image](https://github.com/user-attachments/assets/a76bd631-a63f-4d70-aea0-b88ddb8b176f)

Anyway, new attempt, again trying to set the sensitivity parameter.
[zwave-js-ui-store (3).zip](https://github.com/user-attachments/files/18116575/zwave-js-ui-store.3.zip)

**Reply by kpine** [reply]
The driver logs are there, but they are not **debug** level. While the screenshot is out of date, the textual instructions are not.

> Driver logger can be **configured in Z-Wave section**, **enable Log enabled switch** and **Log To File** and set **Log Level to Debug**

**Reply by Mastiffen** [reply]
Same again, I thought the level was set for everything in General. I have slapped the back of my head three times so you don't have to. 
[zwave-js-ui-store (4).zip](https://github.com/user-attachments/files/18116706/zwave-js-ui-store.4.zip)

**Reply by kpine** [reply]
```text
2024-12-12T19:42:44.952Z SERIAL » 0x011c00133e159f0376006e5991b14eace146fd6b3311a62ddbfa11255811      (30 bytes)
2024-12-12T19:42:44.953Z DRIVER » [Node 062] [REQ] [SendData]
                                  │ transmit options: 0x25
                                  │ callback id:      88
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 118
                                    └─[SupervisionCCGet]
                                      │ session id:      9
                                      │ request updates: true
                                      └─[ConfigurationCCSet]
                                          parameter #:      1
                                          reset to default: false
                                          value size:       1
                                          value format:     SignedInteger
                                          value:            50
2024-12-12T19:42:44.955Z SERIAL « [ACK]                                                                   (0x06)
2024-12-12T19:42:44.962Z SERIAL « 0x0104011301e8                                                       (6 bytes)
2024-12-12T19:42:44.963Z SERIAL » [ACK]                                                                   (0x06)
2024-12-12T19:42:44.963Z DRIVER « [RES] [SendData]
                                    was sent: true
2024-12-12T19:42:45.123Z SERIAL « 0x0107001358000010a3                                                 (9 bytes)
2024-12-12T19:42:45.123Z SERIAL » [ACK]                                                                   (0x06)
2024-12-12T19:42:45.124Z DRIVER « [REQ] [SendData]
                                    callback id:     88
                                    transmit status: OK
2024-12-12T19:42:45.229Z SERIAL « 0x01170004003e119f036e00e604de7901ce3fcb35b4c097d44d                (25 bytes)
2024-12-12T19:42:45.231Z SERIAL » [ACK]                                                                   (0x06)
2024-12-12T19:42:45.232Z DRIVER « [Node 062] [REQ] [ApplicationCommand]
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 110
                                    │ security class:  S2_Unauthenticated
                                    └─[SupervisionCCReport]
                                        session id:          9
                                        more updates follow: false
                                        status:              Fail
                                        duration:            0s
```

The device is reporting a failure. It could be that it doesn't support Supervision CC even though it claims to support it. There are a number of devices like that.

If you want to try you can disable Supervision like: <https://github.com/zwave-js/node-zwave-js/pulls?q=sort%3Aupdated-desc+is%3Apr+disable+supervision+is%3Aclosed+>

Just drop the modified device configuration into the `config` directory of the Z-Wave JS UI, restart ZUI and try again. Not sure if an interview is needed. See https://zwave-js.github.io/zwave-js-ui/#/device-support/custom-device-files.

**Reply by Mastiffen** [reply]
Thanks a lot, I'm sure you're right. So I found a file in the database, but I think "this is not the config file I am looking for". It was here:

[](https://devices.zwave-js.io/?jumpTo=0x0258:0x0200:0x1036:0.0#)

I did not find the word "Supervision" in that, it seems it's just the params. So maye you could please tell me where to find the correct file?

**Reply by kpine** [reply]
You need to **add** the code in the device file to to disable Supervision, that's the potential fix. It doesn't exist. Review any one of the 20+ PRs I posted for examples.

The control panel in ZUI has links to the device page. Click on it for your device. 

![image](https://github.com/user-attachments/assets/be520113-cac9-4a43-95e7-bb6f8c722b5a)

Then from the device page click on "Edit file" and choose the text editor option.

![image](https://github.com/user-attachments/assets/370d12fc-a751-4116-93f4-b7e4b10c4c92)

![image](https://github.com/user-attachments/assets/8dfa7b1c-c71c-4d69-bd8e-9e7d9bd985ae)

Then you can download the file from GitHub and edit it.

**Reply by Mastiffen** [reply]
That's the file I downloaded, only I went via the database, didn't know about that button, thanks! I tried to put the code from one of the other files into it, but I don't see "user -provided device file loaded" in the logs. I do see this:

![image](https://github.com/user-attachments/assets/8bf8fa6b-4137-4eaf-b76c-6698438fd7e3)

I guess that even more points to the same thing, right? This is where I have put the file:

![image](https://github.com/user-attachments/assets/668c8271-dfe9-4ff6-afd5-fce9642a69ce)

Shouldn't that be correct? If there is an error in the file after editing, would that be shown in the log too?

**Reply by kpine** [reply]
> I guess that even more points to the same thing, right? This is where I have put the file:

No, those are Security2 CC messages, not Supervision.

> Shouldn't that be correct? If there is an error in the file after editing, would that be shown in the log too?

Looks correct to me. Make sure you've restarted ZUI for the file to be picked up. If there are any parsing errors they will show up in the console logs.

**Reply by Mastiffen** [reply]
But this means that it's not using the file, right? I restarted for a second time now and didn't see antyhing in the log:

`2024-12-12T20:40:46.585Z CNTRLR   [Node 062] Embedded device config loaded`

**Reply by Mastiffen** [reply]
Here is the json, if you have time to look at it, I added the last bit. I checked it in Notepad++ and I believe I got all the brackets right.

[nas-pd02z-2.json](https://github.com/user-attachments/files/18117345/nas-pd02z-2.json)

**Reply by kpine** [reply]
The log `Embedded device config loaded` means the custom file was not found. Make sure the file is in the correct directory `<store>/config` by default. You showed a screenshot of some directory which by name looks reasonable, but from that I cannot tell whether it's the correct one or not. Is your `Data` directory being mapped to `/usr/src/app/store`?

From a brief inspection I don't see a problem with the json file itself, so I suspect it's just not being found.

**Reply by Mastiffen** [reply]
Thank you very much for your patience! I have it mapped like this:

`    volumes:
      - /media/pi/Docker/Docker-Compose/Z-Wave-JS-UI/Data:/usr/src/app/store
`

I can't imagine that it isn't correct, because every thing else works, and all the other files are in this directory: 

![image](https://github.com/user-attachments/assets/f4f95117-cc81-4796-a657-1fdc65afee88)

Is it possible that there is a capital letter or something somewhere that makes the filename itself  `nas-pd02z-2.json` wrong?

**Reply by kpine** [reply]
The filename doesn't matter (except for maybe `.json` extension). Is your "Config priority dir" setting still the default?

![image](https://github.com/user-attachments/assets/95240466-6016-4f38-8952-06e3c1f870e2)

**Reply by kpine** [reply, marked as answer]
If that last one is not a problem, I only have one more idea.

See these import lines?

https://github.com/zwave-js/node-zwave-js/blob/c80a52d1313a3677d417d81d4e64b5e09ec4ebff/packages/config/config/devices/0x0258/nas-pd02z-2.json#L181-L184

I suspect it's trying to load these from your `config` directory (you might see an error in the Docker logs). Try creating a `config/template` directory and download the [`shenzhen_neo_template.json`](https://raw.githubusercontent.com/zwave-js/node-zwave-js/refs/heads/master/packages/config/config/devices/0x0258/templates/shenzhen_neo_template.json) file into the new template directory, then restart. 

The imports with the tilde in the path (e.g. `~/templates/master_template.json`) should be OK as-is.

**Reply by Mastiffen** [reply]
You posted the second one while I was writing! Damned good catch about the directory, it was the one from before I went Docker! Now you are really getting somewhere (I'm just the monkey at the keyboard right now).  And the message is indeed what  you thought:

`2024-12-13T07:27:49.089Z CONFIG   Error parsing config file nas-pd02z-2.json: Could not parse config file /usr/s
                                  rc/app/store/config/templates/shenzhen_neo_template.json: ENOENT: no such file
                                   or directory, open '/usr/src/app/store/config/templates/shenzhen_neo_template
                                  .json'
                                  Import stack: 
                                    in #blink_on_motion
                                    in /usr/src/app/store/config/nas-pd02z-2.json (ZW0340)`

There is no directory called config/templates, I created it now and downloaded the template. And the orange "Auto discovery functionality has changed" reinterview button came up! The interview was successfull, and the value updated!!!! Thank you again very much for your help and patience! Have a brilliant weekend!

**Reply by kpine** [reply]
You can fix this by also modifying the import statements (https://github.com/zwave-js/node-zwave-js/pull/6067) to fallback to the embedded manufacturer template. This rang a bell because it was a feature I requested. :laughing: 

Just change:

```json
 { 
 	"#": "7", 
 	"$import": "templates/shenzhen_neo_template.json#blink_on_motion" 
 }, 
```
to:
```json
 { 
 	"#": "7", 
 	"$import": "~/0x0258/templates/shenzhen_neo_template.json#blink_on_motion" 
 }, 
```

etc., and you should be able to delete your local copy of `shenzhen_neo_template.json`.

**Reply by Mastiffen** [reply]
Nice to revisit old requests? 😉 But it seems that there is an error either in the change or the file name. Am I correct in assuming that if it was a line in the file it would refer to that line, or is that off? I saved a copy of everything before I started, as usual.

`2024-12-13T07:52:14.008Z CONFIG   Error parsing config file templatesbak/shenzhen_neo_template.json: The device 
                                  config file is malformed!
                                  packages/config/config/devices/templatesbak/shenzhen_neo_template.json:
                                  manufacturer id must be a lowercase hexadecimal number with 4 digits (ZW0340)`

I changed the file name to 0x0258.json, but got the same error. And still: 

`2024-12-13T07:56:36.996Z CNTRLR   [Node 062] User-provided device config loaded
2024-12-13T07:56:37.002Z CNTRLR   [Node 063] User-provided device config loaded`

Edit: But I see in the actual device that it isn't parsed correctly, because it's mostly "vundefined" there now.

**Reply by Mastiffen** [reply]
Here is the file [0x0258.json](https://github.com/user-attachments/files/18123294/0x0258.json).

It's probably a "monkey at the keyboard" error...

**Reply by Mastiffen** [reply]
And it seems like the parameter is wrong, even if I can set it. It doesn't take, and "get parameter" will say "Undefined" on both PIR sensors even right after I send it and get "updated". I wonder what the problem with this thing is. Another thing is that only one of the two are totally crazy in showing motion and no motion, the other one works. So I guess there is even another thing here, a faulty sensor.

Edit: And even with the motion sensoring working in Hass for the one that isn't damaged, it shows "true" constantly in `[63-48-0-Motion] Sensor state (Motion)`, which it didn't before. Weird stuff...

**Reply by AlCalzone** [reply]
> 2024-12-13T07:52:14.008Z CONFIG Error parsing config file templatesbak/shenzhen_neo_template.json: The device config file is malformed! packages/config/config/devices/templatesbak/shenzhen_neo_template.json: manufacturer id must be a lowercase hexadecimal number with 4 digits (ZW0340)

The error message is bad, but every directory directly inside `devices` that's not exactly called `templates` is considered to contain valid device config files, which the template isn't. Put the `templatesbak` directory inside the `0x0258` directory and correct the import path, then the error should be gone.



> And it seems like the parameter is wrong, even if I can set it. It doesn't take, and "get parameter" will say "Undefined" on both PIR sensors even right after I send it and get "updated".

That's one of the problems with those chinese devices. You sometimes get a manual that belongs to a different firmware, parameter documentation is sometimes just wrong, or seemingly identical devices have different firmware inside with different behavior.
I can verify if you post a driver log of that GET attempt, but I assume the device doesn't actually have the parameter you're trying to read.

**Reply by Mastiffen** [reply]
Sorry, double monkey at the keyboard eror on the directory. That screenshot was after I renambed it from templates to take it out of the picture. I thought it had to be underneath config/templates. I first thought you meant like this:

![image](https://github.com/user-attachments/assets/32848bda-cd40-4c5e-ad13-aec7d5b64806)

But I think I misunderstood that one too, because after a reinterview nothing is shown in the device, except for configuration, which only has the custom configuration.

**Reply by AlCalzone** [reply]
Share a driver log of the startup of Z-Wave JS UI and the following re-interview

**Reply by Mastiffen** [reply]
Here. Did a new restart and a new reinterview, to avoid all the clutter that had come in the meantime.
[zwave-js-ui-store (5).zip](https://github.com/user-attachments/files/18129538/zwave-js-ui-store.5.zip)
